app/lib/attachment_batch.rb

```diff
@@ -77,10 +77,16 @@ def remove_files
           when :fog
             logger.debug { "Deleting #{attachment.path(style)}" }
 
+            retries = 0
             begin
               attachment.send(:directory).files.new(key: attachment.path(style)).destroy
-            rescue Fog::Storage::OpenStack::NotFound
-              # Ignore failure to delete a file that has already been deleted
+            rescue Fog::OpenStack::Storage::NotFound
+              logger.debug "Will ignore because file is not found #{attachment.path(style)}"
+            rescue => e
+                sleep(5)
+                retry if (retries += 1) < MAX_RETRY
+                logger.error "Batch deletion from fog failed after #{e.message}"
+                raise e
```

Comment:
This is indented too deep and should be fixed.

Additionally, I'd follow the same style as for the retry logic further down that method, for consistency, so something like:
```suggestion
              retries += 1

              if retries < MAX_RETRY              
                logger.debug "Retry #{retries}/#{MAX_RETRY} after #{e.message}"
                sleep 5
              else
                logger.error "Batch deletion from fog failed after #{e.message}"
                raise e
              end
```

Reply:
Makes sense. Fixed and changed. I followed exactly the same logic and made `sleep 2**retries` too.